### ISSUE TITLE
resourceParameters trigger for sources that weren't fired, overwriting other resourceParameters

### ISSUE THREAD

**Issue by jaygorrell** [argoproj/argo-events#365]
**Describe the bug**
I have a calendar+sensor where the calendar contains many crons that each pass a different parameter to the same sensor/template. The sensor has a `resourceParameters` list where each cron source maps it's `userPayload` into a field.

The issue is that each mapping is evaluated on every trigger. So I need to put a default on each `src` in the list... which is fine. But then, the last one in the list will use that default as the value for the template. No matter which one was triggered.

**To Reproduce**
1. Create calendar with 3 sources, each with a different schedule and `userPayload`.
2. Create a sensor that listens to these sources.
3. Configure dependencyGroups and a circuit so that each of the 3 can trigger the workflow independently (`g1 || g2 || g3`).
4. Use `resourceParameters` to map `userPayload` of each source to something in the workflow.
5. See that only the last listed mapping matters.

I think the key is that `resourceParameters` requires that when the sensor is triggered, all values can be evaluated from the payload. Since only one of the sources is triggered, the others need default values -- but that default actually takes effect. 

**Expected behavior**
Only the matched source should trigger `resourceParameters` replacement.

**Environment (please complete the following information):**
 - Version 0.9.3

Is there a different way to do this that I'm missing?

**Comment by l15k4** [comment]
I have a very similar possible even the same issue. Imho this is a BUG in logic, imagine this scenario : 
having dependencyGroups with `any` trigger : 
```
        when:
          any:
            - "foo"
            - "bar"
            - "baz"
```
and corresponding `resourceParameters` : 
```
      resourceParameters:
        - src:
            event: foo-event-source
          dest: spec.arguments.parameters.0.value
        - src:
            event: bar-event-source
          dest: spec.arguments.parameters.0.value
        - src:
            event: baz-event-source
          dest: spec.arguments.parameters.0.value
```
where events are the same `{"eventTime":"2019-10-21T19:10:00.000Z"}`, they are just coming from a different `source`. One would expect the workflow would be triggered with the event from `any` source, but what actually happens is that when any event comes, errors is thrown : 
```
INFO[2019-10-25 20:05:40] all event dependencies are marked completed, processing triggers 
WARN[2019-10-25 20:05:40] event in event dependency does not exist, cannot apply parameter  param-dest=spec.arguments.parameters.0.value param-src="webhook-gateway:bar"
WARN[2019-10-25 20:05:40] event in event dependency does not exist, cannot apply parameter  param-dest=spec.arguments.parameters.0.value param-src="scheduling-gateway:baz"
ERRO[2019-10-25 20:05:40] trigger failed to execute                     error="failed to apply params. err: trigger failed to execute                     error="failed to apply params. err: unable to resolve 'webhook-gateway:bar' parameter value. verify the path: '' is valid and/or set a default value for this param" trigger-name=hello-trigger
```
So imho currently it is not possible to use `any`, it will imho always blow up this way because as many `any` you have, as many `src` there should be, and this error always happens.

**Comment by l15k4** [comment]
Ok I might be completely wrong, imho `circuit` with `any` is quite broken in `0.10` . I cannot reverse engineer how an event of corresponding source is passed to the workflow. It blows up no matter what I try.

**Comment by l15k4** [comment]
Imho exctly this example of the complex circuit does not work now in `0.10` 
https://argoproj.github.io/argo-events/sensor/#circuit as there is : 
```
        when:
          any:
            - "group_1"
            - "group_2"
```
and just a single `src` : 
```
      resourceParameters:
        - src:
            event: "webhook-gateway-http:endpoint1"
          dest: spec.arguments.parameters.0.value
```

This would blow up on `event` from `"webhook-gateway-http:endpoint2` ... It only works for event from `"webhook-gateway-http:endpoint1`

**Comment by VaibhavPage** [comment]
@l15k4  This definitely looks like a bug to me. I will take a closer look at the code.

**Comment by VaibhavPage** [comment]
Fix is available in v0.12
